apps/hexagon_dma/pipeline_nv12.cpp

```diff
@@ -15,12 +16,16 @@ class DmaPipeline : public Generator<DmaPipeline> {
         // multiply update in tiles.
         Func copy_y("copy_y");
         Func copy_uv("copy_uv");
+        Func uv_de_interleaved("uv_de_interleaved");
+        Func processed_uv("processed_uv");
 
-        copy_y(x, y, c) = input(x, y, c);
-        copy_uv(x, y, c) = input(x, y, c);
+        copy_y(x, y) = input_y(x, y);
+        copy_uv(x, y, c) = input_uv(x, y, c);
+        uv_de_interleaved(x, y, c) = copy_uv(x/2*2, y/2, c);
```

Comment:
Add parens to make this more clear: `(x/2)*2`

That said, assuming we're building the "ideal" thing here (without backend changes yet, per your PR comment), this pipeline should look like this:

        copy_y(x, y) = input_y(x, y);
        copy_uv(x, y, c) = input_uv(x, y, c);
 
        output_y(x, y) = copy_y(x, y) * 2;
        output_uv(x, y, c) = copy_uv(x, y, c) * 2;

Because the DMA engine will need the UV buffer to be interleaved, you will need to do this in the schedule:

        // For internal pipeline buffers, we can just reorder the storage so c is innermost, matching the DMA engine expectations
        // (this is an interleaved buffer).
        copy_uv.reorder_storage(c, x, y)

        // For input/output buffers, we need to tweak the expected extents/strides:
        input_uv.dim(0).set_stride(2);
        input_uv.dim(2).set_stride(1).set_extent(2);
        output_uv.output_buffer().dim(0).set_stride(2);
        output_uv.output_buffer().dim(2).set_stride(1).set_extent(2);